### DISCUSSION TITLE
IPv6 Only EKS Clusters

### DISCUSSION THREAD

**Discussion by sandy-0007** [cilium/cilium#35633] [General]
Hello,

I'm looking to explore on full IPv6 only EKS clusters without dual-stack worker nodes. So I was referring to the [quick installation guide](https://docs.cilium.io/en/stable/gettingstarted/k8s-install-default/) for Cilium CLI.

However, I got stumped by the below limitation

"The AWS ENI integration of Cilium is currently only enabled for IPv4. If you want to use IPv6, use a datapath/IPAM mode other than ENI."

Do you have any recommendation on "use a datapath/IPAM mode other than ENI."?

**Comment by joestringer** [comment]
As of Cilium 1.16, I don't think Cilium supports IPv6-only environments using IPAM from AWS so if you wanted Pods to have IPs that are routable in EKS this would be a new feature request.

The proposal for "use a datapath/IPAM mode other than ENI" is proposing to use one of the other IP address management modes: https://docs.cilium.io/en/stable/network/concepts/ipam/#ip-address-management-ipam
